akka-http-core/src/main/scala/akka/http/scaladsl/ConnectionContext.scala

```diff
@@ -28,18 +30,29 @@ object ConnectionContext {
     enabledCipherSuites: Option[immutable.Seq[String]] = None,
     enabledProtocols:    Option[immutable.Seq[String]] = None,
     clientAuth:          Option[TLSClientAuth]         = None,
-    sslParameters:       Option[SSLParameters]         = None) =
-    new HttpsConnectionContext(sslContext, sslConfig, enabledCipherSuites, enabledProtocols, clientAuth, sslParameters)
+    sslParameters:       Option[SSLParameters]         = None,
+    http2:               UseHttp2                      = UseHttp2.Negotiated) =
+    new HttpsConnectionContext(sslContext, sslConfig, enabledCipherSuites, enabledProtocols, clientAuth, sslParameters, http2)
   //#https-context-creation
 
+  // for binary-compatibility, since 1.10.2
+  def https(
+    sslContext:          SSLContext,
+    sslConfig:           Option[AkkaSSLConfig],
+    enabledCipherSuites: Option[immutable.Seq[String]],
+    enabledProtocols:    Option[immutable.Seq[String]],
+    clientAuth:          Option[TLSClientAuth],
+    sslParameters:       Option[SSLParameters]) =
+    new HttpsConnectionContext(sslContext, sslConfig, enabledCipherSuites, enabledProtocols, clientAuth, sslParameters, http2 = Negotiated)
+
   // for binary-compatibility, since 2.4.7
   def https(
     sslContext:          SSLContext,
     enabledCipherSuites: Option[immutable.Seq[String]],
     enabledProtocols:    Option[immutable.Seq[String]],
     clientAuth:          Option[TLSClientAuth],
     sslParameters:       Option[SSLParameters]) =
-    new HttpsConnectionContext(sslContext, None, enabledCipherSuites, enabledProtocols, clientAuth, sslParameters)
+    new HttpsConnectionContext(sslContext, sslConfig = None, enabledCipherSuites, enabledProtocols, clientAuth, sslParameters, http2 = Negotiated)
```

Comment:
Not *sure* this is the correct default yet (between `Never` and `Negotiated`), but right now there's no functional difference yet, so we can still revisit this.

Reply:
Negotiated is ok